### PR TITLE
Bot: create tutorialhell command

### LINKED ISSUE #462: /multiplecourses - command to address questions about supplementing top concurrently
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)

-   [x] The title of this issue follows the `command name: brief description of request` format, e.g. `/help: add optional @user parameter`

The following checkbox is OPTIONAL:

-   [x] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**

Feature:
- Slash command /multiplecourses that contain Briggs's advice about the topic.

Benefit:
- Reduce friction to provide learners with what TOP recommends regarding topic
- Learners get consistent messaging.

Solves:
- Having to search for a pinned message, or taking the time to type out a response in the moment that paraphrases the advice contained in the pin.
- Less workload for those involved in moderating or even helping out in chat.

**2. Acceptance Criteria:**

- [ ] /multiplecourses typed in textbox would bring up the command
- [ ] Optional ability to tag a user 
- [ ] Contains the content:
https://discord.com/channels/505093832157691914/505093832157691916/778727680438698055 
https://discord.com/channels/505093832157691914/505093832157691916/778727683675914300
https://discord.com/channels/505093832157691914/505093832157691916/778728005904105512
(perhaps edited for brevity)

**3. Additional Information:**

Open to a different command name. It's just what came to me.

### PR DESCRIPTION
**Pull request by JustWaveThings**
## Because

Benefit:

- Reduce friction to provide learners with what TOP recommends regarding topic
- Learners get consistent messaging.

Solves:

- Having to search for a pinned message, or taking the time to type out a response in the moment that paraphrases the advice contained in the pin.
- Less workload for those involved in moderating or even helping out in chat.

## This PR

- creates /tutorialhell command
- optional user ping
- contains edited content discussed in issue

## Issue

Closes #462 

## Additional Information

- Summarized Briggs's messages on the topic, and added Carlos's message about picking a course and sticking with it, to be more directive for those who need that, and a bit more course agnostic. If someone feels that FCC or whatever is the best course for them, I am happy for them.

- I'm not in love with the numbering, but I felt like it might be nice to say to a user 'read number 2 in particular'.

- It's still a lot of content for one command. I shrunk Briggs's message by over 50%. I think the synergy of the points is important though, so unless a more succinct rewrite is possible, this works.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Callbacks command: Update verbiage`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If this PR adds new features or functionality, I have added new tests
-   [x] If applicable, I have ensured all tests related to any command files included in this PR pass, and/or all snapshots are up to date

### CODE REVIEW

File: new-era-commands/slash/tutorialHell.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      option.setName("user").setDescription("user to ping")
+    ),
+  execute: async (interaction) => {
+    const userId = interaction.options.getUser("user");
+    const tutorialHellEmbed = new EmbedBuilder()
+      .setColor("#cc9543")
+      .setTitle("TOP's advice on doing more than one course at a time")
+      .setDescription(`
+1. There is no "best" course when you're a self-learner. It's about learning yourself and not being handed the concepts. 
+2. Beginner content dominates learning resources, but excessive topic repetition, known as 'Tutorial Hell,' isn't helpful. Instead, focus on practice and progress. Concepts often become clearer over time. Using multiple courses simultaneously prolongs basic learning, by lacking the understanding to integrate the concepts fully. This wastes time. Switching courses can lead to context loss and increased repetition, further slowing you down. After learning, reinforce with practice, not mere topic reiteration.
```
Comment (ManonLef) [inline new-era-commands/slash/tutorialHell.js:17]: I think this could be simplified to come across stronger. The basic message is quite strong. 

> "Using multiple courses simultaneously prolongs basic learning, by lacking the understanding to integrate the concepts fully."

I think this might be missing emphasis on the concept of "doing multiple courses basically means you're spending more time on repeating the basics". So we might be focusing on the result here instead of the cause of "increased repetition" mentioned later, but thus possibly losing strength.

File: new-era-commands/slash/tutorialHell.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+3. Different courses, different methods. Mixing them can confuse you and scatter your attention. This could extend your learning time or leave you with shallow knowledge. Following one course eliminates this.
+4. Finding help on topics in courses with a weak support community is difficult, and shifting this burden to another is rude and unlikely to be helpful.
+5. Undecided? Pick the course you feel you are most likely to complete.
```
Comment (ManonLef) [inline new-era-commands/slash/tutorialHell.js:20]: I think this deserves the number one spot. And it's also the one I think deserves to be more lengthy since the best course is the one you stick to.

File: new-era-commands/slash/tutorialHell.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    ),
+  execute: async (interaction) => {
+    const userId = interaction.options.getUser("user");
+    const tutorialHellEmbed = new EmbedBuilder()
+      .setColor("#cc9543")
+      .setTitle("TOP's advice on doing more than one course at a time")
+      .setDescription(`
+1. There is no "best" course when you're a self-learner. It's about learning yourself and not being handed the concepts. 
+2. Beginner content dominates learning resources, but excessive topic repetition, known as 'Tutorial Hell,' isn't helpful. Instead, focus on practice and progress. Concepts often become clearer over time. Using multiple courses simultaneously prolongs basic learning, by lacking the understanding to integrate the concepts fully. This wastes time. Switching courses can lead to context loss and increased repetition, further slowing you down. After learning, reinforce with practice, not mere topic reiteration.
+3. Different courses, different methods. Mixing them can confuse you and scatter your attention. This could extend your learning time or leave you with shallow knowledge. Following one course eliminates this.
```
Comment (ManonLef) [inline new-era-commands/slash/tutorialHell.js:18]: I think this can be combined with 2.

File: new-era-commands/slash/tutorialHell.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+4. Finding help on topics in courses with a weak support community is difficult, and shifting this burden to another is rude and unlikely to be helpful.
```
Comment (ManonLef) [inline new-era-commands/slash/tutorialHell.js:19]: this feels a bit off-topic compared to the other items. I do see why you mention it I think, we could add it as a tip that if you're having a rough time deciding, looking at community support might help with the decision? "Shifting this burden" is something that feels a bit unexplained to me and I'm not sure we really want to go into it too much here since the focus should be on "different courses" in my opinion, unless you also want to tackle the problem of asking help here with other courses.

**Review by ManonLef — COMMENTED** [review]
Thanks so much for this @JustWaveThings ! I really think a command like this would be valuable! It's definitely a topic coming up a lot. 

I do think that people are less likely to read large amounts of text so I'd love it if we could look at possible ways to extract the main message from each point, communicate that first and only expand where needed. 

As I see it now I think you wish to bring across these basics, but please do correct me if I am wrong, and also please do tell me if you think we shouldn't decrease the amount of words :)

1: The best course is the one you stick to (combined with 5) and doesn't hold your hand
2: repeating beginner content endlessly, which is an obvious side effect of doing multiple courses, is not productive and often leads to "tutorial hell"
3: confusion with different learning methods
4: community (although I'm feeling this is the one standing out)
5: pick and stick. 

I hope the amount of comments doesn't put you off. I personally feel strongly about "short and sweet" if we can, and I think we might make the message stronger if we manage to settle on the key points.

File: new-era-commands/slash/tutorialHell.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    .addUserOption((option) =>
+      option.setName("user").setDescription("user to ping")
+    ),
+  execute: async (interaction) => {
+    const userId = interaction.options.getUser("user");
+    const tutorialHellEmbed = new EmbedBuilder()
+      .setColor("#cc9543")
+      .setTitle("TOP's advice on doing more than one course at a time")
+      .setDescription(`
+1. There is no "best" course when you're a self-learner. It's about learning yourself and not being handed the concepts.
```
[the remark this answers is not in the document]
Comment (JustWaveThings) [reply new-era-commands/slash/tutorialHell.js:16]: I agree with replacing

> It's about learning yourself and not being handed the concepts. 

with 

> It's about what you're putting into it"

File: new-era-commands/slash/tutorialHell.js (hunk 2 above)
Comment (JustWaveThings) [reply new-era-commands/slash/tutorialHell.js:20]: I like having this one last, as the other points explain why multiple courses are bad, and this point subtly suggests that the best course for you is the one you complete, not the objectively better one that you give up on.

File: new-era-commands/slash/tutorialHell.js (hunk 3 above)
Comment (JustWaveThings) [reply new-era-commands/slash/tutorialHell.js:18]: I think three stands on its own as learning competing methods/tooling/paradigms can make it feel like you are learning more than you really are, where point two describes problems with increased repetition.

File: new-era-commands/slash/tutorialHell.js (hunk 4 above)
Comment (JustWaveThings) [reply new-era-commands/slash/tutorialHell.js:19]: I see what you mean. It could benefit from a rephrasing. I could approach this point as a person evaluating their options, to consider the type and availability of community support.

Also, I would remove this:

>  and shifting this burden to another is rude and unlikely to be helpful.

File: new-era-commands/slash/tutorialHell.js (hunk 1 above)
Comment (JustWaveThings) [reply new-era-commands/slash/tutorialHell.js:17]: Agreed.  I can shorten that and come out ahead.

File: new-era-commands/slash/tutorialHell.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (JustWaveThings) [reply new-era-commands/slash/tutorialHell.js:16]: I agree with updating 
>  It's about learning yourself and not being handed the concepts.
to 
> It's about what you're putting into it.

**Comment by JustWaveThings** [comment]
Thank you so much for the feedback, Manon! 

I can make the changes discussed in the comments above and we can roll with that.

Or, since this PR has already shaved off 50% of Briggs's original word count, and if we are concerned about it being too wordy still for a bot command, I wonder if it wouldn't just be better to have a bulleted list somewhat like you have and link to the original posts. I don't want to stray too far from what Briggs wrote.

Example: 

> Heading: Multiple Courses / Avoid Tutorial Hell
>
> Reasons why we recommend doing one course versus multiple:
>
> >  1: The best course is the one you stick to (combined with 5) and doesn't hold your hand
>  2: repeating beginner content endlessly, which is an obvious side effect of doing multiple courses, is not productive and 
>  often leads to "tutorial hell"
> 3: confusion with different learning methods
> 4: community (although I'm feeling this is the one standing out)
> 5: pick and stick.
> 
> See the posts that expand on this list <links to original message here, here, and here> 

Which of these options do you think fits what we'd want for a /tutorialhell command?

**Comment by JustWaveThings** [comment]
Is this something we still want to pursue?

**Comment by 01zulfi** [comment]
@JustWaveThings Let's get a final say on Discord regarding verbiage, and then merge.

**Review by 01zulfi — APPROVED** [review]
lgtm :rocket:
